java-checks-test-sources/src/main/java/checks/CompareStringsBoxedTypesWithEqualsCheck.java

```diff
@@ -1,58 +1,117 @@
 package checks;
 
-class Boxed {
-  String str1 = "blue";
-  String str2 = "blue";
-  Integer boxedInt1 = 1;
-  Integer boxedInt2 = 2;
-  int myInt1 = 1;
-  int myInt2 = 2;
-  private void method() {
-    if (str1 == str2) {} // Noncompliant
-    if (str1 == "blue") {} // Noncompliant
-    if (boxedInt1 == boxedInt2) {} // Noncompliant
-    if (myInt1 == myInt2) {}
-    if (boxedInt1 == null) {}
-
-    if (str1 != str2) {} // Noncompliant
-    if (str1 != "blue") {} // Noncompliant
-    if (boxedInt1 != boxedInt2) {} // Noncompliant
-    if (myInt1 != myInt2) {}
-    if (boxedInt1 != null) {}
-
-    if (boxedInt1 == myInt1) {} // Compliant: unboxing conversion
-    if (boxedInt1 != myInt1) {} // Compliant: unboxing conversion
-
-    if (boxedInt1 > boxedInt2) {}
-    if (null != str1) {}
-
-    if (str1.equals(str2)) {}
-    if (boxedInt1.equals(boxedInt2)) {}
-    
-    Boolean b = null;
-    Boolean b1 = true;
-    Boolean b2 = false;
-    Boolean b3 = Boolean.TRUE;
-    Boolean b4 = Boolean.FALSE;
-    Boolean b5 = new Boolean(true);
-    Boolean b6 = new Boolean(false);
-    if (b == Boolean.FALSE) {} // Compliant
-    if (b1 == Boolean.TRUE) {} // Compliant
-    if (b2 == Boolean.TRUE) {} // Compliant
-    if (b3 == Boolean.FALSE) {} // Compliant
-    if (b4 == Boolean.TRUE) {} // Compliant
-    if (Boolean.TRUE == b4) {} // Compliant
-    if (b5 == Boolean.TRUE) {} // FN
-    if (b5 == b6) {} // Noncompliant
-  }
-}
+import java.util.Locale;
+
+class CompareStringsBoxedTypesWithEqualsCheck {
+  class Boxed {
```

Comment:
Instead of nesting the classes into a single one, you could rename directly the three classes to keep the meta data of the changes.

Reply:
I am not sure I understand this comment

Reply:
When looking at the diff, the whole file will appear as new code, while you only modified part of it. To avoid this, you should not nest everything into another class.

Reply:
As discussed, the main point is to avoid name clashes in the test module, therefore to not have names like "Boxed" at the top level. Both the current code and my suggestion do the trick. which one is not so important in the end, it is fine as it is.